Revit_Core_Engine/Convert/Environment/ToRevit/CableTray.cs

```diff
@@ -0,0 +1,105 @@
+namespace BH.Revit.Engine.Core
+{
+    public static partial class Convert
+    {
+        /***************************************************/
+        /****              Public methods               ****/
+        /***************************************************/
+
+        // ToDo for method functionality from top to bottom: 
+        // SectionProperty.ToRevitElementType to BuiltInCategory is unknown and currently set to OST_CableTrayRun
+
+        public static Autodesk.Revit.DB.Electrical.CableTray ToRevitCableTray(this oM.MEP.System.CableTray cableTray, Document document, RevitSettings settings = null, Dictionary<Guid, List<int>> refObjects = null)
+        {
+            if (document == null)
+                return null;
+
+            // Check valid cableTray object
+            if (cableTray == null)
+                return null;
+
+            // Construct Revit Cable Tray
+            Autodesk.Revit.DB.Electrical.CableTray revitTray = refObjects.GetValue<Autodesk.Revit.DB.Electrical.CableTray>(document, cableTray.BHoM_Guid);
+            if (revitTray != null)
+                return revitTray;
+
+            // Settings
+            settings = settings.DefaultIfNull();
+
+            // CableTray type
+            Autodesk.Revit.DB.Electrical.CableTrayType trayType = cableTray.SectionProperty.ToRevitElementType(document, new List<BuiltInCategory> { BuiltInCategory.OST_CableTrayRun }, settings, refObjects) as Autodesk.Revit.DB.Electrical.CableTrayType;
+
+            // End points
+            XYZ start = cableTray.StartPoint.ToRevit();
+            XYZ end = cableTray.EndPoint.ToRevit();
+
+            // Level
+            Level level = document.LevelBelow(Math.Min(start.Z, end.Z), settings);
+            if (level == null)
+                return null;
+
+            // Copy parameters from BHoM object to Revit element
+            revitTray.CopyParameters(cableTray, settings);
+
+            // Set Orientation angle
+            double orientationAngle = cableTray.OrientationAngle;
+            if (Math.Abs(orientationAngle) > settings.AngleTolerance)
+            {
+                ElementTransformUtils.RotateElement(document, revitTray.Id, Line.CreateBound(start, end), orientationAngle);
+            }
+
+            SectionProfile sectionProfile = cableTray.SectionProperty?.SectionProfile;
+
+            BoxProfile elementProfile = sectionProfile.ElementProfile as BoxProfile;
+            if (elementProfile == null)
+                return null;
+
+            // Set Height
+            double profileHeight = elementProfile.Height;
+            revitTray.SetParameter(BuiltInParameter.RBS_CABLETRAY_HEIGHT_PARAM, profileHeight);
+
+            // Set Width
+            double profileWidth = elementProfile.Width;
+            revitTray.SetParameter(BuiltInParameter.RBS_CABLETRAY_WIDTH_PARAM, profileWidth);
+
+            revitTray = Autodesk.Revit.DB.Electrical.CableTray.Create(document, trayType.Id, start, end, level.Id);
```

Comment:
you are first trying to set parameters and rotate the tray and then you create it - that is not going to work 😃 

By the looks of it, this line should go before line 74, preceded by lines currently being 85-87. You also need to null check after creation.